amd_openvx_extensions/amd_rpp/source/tensor/Transpose.cpp

```diff
@@ -0,0 +1,296 @@
+    }    
+    if (numDims == 4) {
+        RpptROI *src_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            unsigned index = i * 3 * 2;
+            if (data->inputLayout == vxTensorLayout::VX_NHWC) {
+                data->pSrcRoi[index + 0] = src_roi[i].xywhROI.xy.y;
+                data->pSrcRoi[index + 1] = src_roi[i].xywhROI.xy.x;
+                data->pSrcRoi[index + 2] = 0;
+                data->pSrcRoi[index + 3] = src_roi[i].xywhROI.roiHeight;
+                data->pSrcRoi[index + 4] = src_roi[i].xywhROI.roiWidth;
+                data->pSrcRoi[index + 5] = data->inputTensorDims[3];
+            } else if (data->inputLayout == vxTensorLayout::VX_NCHW) {
+                data->pSrcRoi[index + 0] = 0;
+                data->pSrcRoi[index + 1] = src_roi[i].xywhROI.xy.y;
+                data->pSrcRoi[index + 2] = src_roi[i].xywhROI.xy.x;
+                data->pSrcRoi[index + 3] = data->inputTensorDims[3];
+                data->pSrcRoi[index + 4] = src_roi[i].xywhROI.roiHeight;
+                data->pSrcRoi[index + 5] = src_roi[i].xywhROI.roiWidth;
+            }
+        }
+    } else {
+        data->pSrcRoi = static_cast<unsigned *>(roi_tensor_ptr);
+    }
+    return status;
+}
+
+static vx_status VX_CALLBACK validateTranspose(vx_node node, const vx_reference parameters[], vx_uint32 num, vx_meta_format metas[]) {
+    vx_status status = VX_SUCCESS;
+    vx_enum scalar_type;
+
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[4], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #4 type=%d (must be size)\n", scalar_type);
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[5], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #5 type=%d (must be size)\n", scalar_type);
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[6], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #6 type=%d (must be size)\n", scalar_type);
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[7], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_UINT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #7 type=%d (must be size)\n", scalar_type);
+
+    // Check for input parameters
+    size_t num_tensor_dims;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &num_tensor_dims, sizeof(num_tensor_dims)));
+    if(num_tensor_dims < 3) return ERRMSG(VX_ERROR_INVALID_DIMENSION, "validate: Transpose: tensor: #0 dimensions=%lu (must be greater than or equal to 4)\n", num_tensor_dims);
```

Comment:
Must be greater than or equal to 4 or 3? Please check and change wrt the condition